pkg/controller/daemon/controller.go

```diff
@@ -383,22 +384,35 @@ func (dsc *DaemonSetsController) manage(ds *extensions.DaemonSet) {
 	dsc.expectations.SetExpectations(dsKey, len(nodesNeedingDaemonPods), len(podsToDelete))
 
 	glog.V(4).Infof("Nodes needing daemon pods for daemon set %s: %+v", ds.Name, nodesNeedingDaemonPods)
+	createWait := sync.WaitGroup{}
+	createWait.Add(len(nodesNeedingDaemonPods))
 	for i := range nodesNeedingDaemonPods {
-		if err := dsc.podControl.CreatePodsOnNode(nodesNeedingDaemonPods[i], ds.Namespace, ds.Spec.Template, ds); err != nil {
-			glog.V(2).Infof("Failed creation, decrementing expectations for set %q/%q", ds.Namespace, ds.Name)
-			dsc.expectations.CreationObserved(dsKey)
-			util.HandleError(err)
-		}
+		go func(ix int) {
```

Comment:
@mqliang looking better!! I agree with @smarterclayton though and there should be a limiting factor on the amount of goroutines that are started so that there is not a chance of bursting a massive amount (not likely but possible).